src/staging/validation.py

```diff
@@ -211,10 +211,74 @@ def validate_data_with_schema(survey_df: pd.DataFrame, schema_path: str):
             # Replace whatever is in that column with np.nan
             survey_df[column] = nan
             dtypes_dict[column] = "float64"
+
+        try:
+            validation_logger.debug(f"{column} before: {survey_df[column].dtype}")
+            if dtypes_dict[column] == "Int64":
+                # Convert non-integer string to NaN
+                survey_df[column] = survey_df[column].apply(
+                    pd.to_numeric, errors="coerce"
+                )
+                # Cast columns to Int64
+                survey_df[column] = survey_df[column].astype(pd.Int64Dtype())
+            else:
+                survey_df[column] = survey_df[column].astype(dtypes_dict[column])
+            validation_logger.debug(f"{column} after: {survey_df[column].dtype}")
+        except Exception as e:
+            validation_logger.error(e)
+
+
+def validate_data_with_both_schema(
+    survey_df: pd.DataFrame, schema_path1: "str", schema_path2: "str"
+):
+    """Takes the schemas from the toml file and validates the survey data df.
+
+    Args:
+        survey_df (pd.DataFrame): Survey data in a pd.df format
+        schema_path1 (str): path to the schema toml (should be in config folder)
+        schema_path2 (str): path to the schema toml (should be in config folder)
```

Comment:
Can you be specific with the names here? Which schema are you expecting for 1 and 2? You can be specific in the name and the description.